File: Java/LECANDY_Little-Elephant-and-Candies.java
import java.util.Scanner;
import java.util.stream.IntStream;

class LittleElephantAndCandies {
	public static void main(String[] args) {
		Scanner scanner = new Scanner(System.in);
		int testCase = scanner.nextInt();
		while (testCase-- != 0) {
			int numberOfElephants = scanner.nextInt();
			int numberofCandies = scanner.nextInt();
			int[] candyNeeded = new int[numberOfElephants];
			for (int i = 0; i < numberOfElephants; i++) {
				candyNeeded[i] = scanner.nextInt();
			}

			int candyRequired = IntStream.of(candyNeeded).sum();
			if (candyRequired > numberofCandies)
				System.out.println("No");
			else
				System.out.println("Yes");
		}
	}
}

File: C++/LECANDY_Little-Elephant-and-Candies.c
#include <iostream>
#include <vector>
#include <numeric>

using namespace std;

int main()
{
    int testCase;
    cin >> testCase;

    while(testCase--){
        int n,c;
        cin>> n >> c;

        vector <int> a(n);
        for(int i = 0; i < n; i++){
            cin>>a[i];
        }

        int sum = accumulate(a.begin(), a.end(), 0);

        if(sum > c)
            cout<<"No"<<endl;
        else
            cout<<"Yes"<<endl;

     }
    return 0;
}
